docs/developer-guide.md

```diff
@@ -185,52 +185,152 @@ TestUtils.mock_postal_code()
 
 #### Run Playwright Specs
 
-1.0 Ensure the server is running:
+Certainly! Below is a simple template for documentation on writing Playwright tests, including best practices and tips. Feel free to customize it based on your specific needs:
+
+## 1.0 Ensure the server is running:
 
 Start server from new terminal command:
 
+# Playwright Testing Documentation
+
+## Introduction
+
+Playwright is a powerful end-to-end testing library for web applications. It provides a simple API to automate browser actions, allowing developers to create robust and maintainable tests.
+
+## Getting Started
+
+To get started with Playwright, you'll need to install it as a dependency in your project:
+
 ```bash
 cd bc_obps && make run
+npm install @playwright/test
 ```
 
 2.0 Ensure the client app is running:
 
-Start client app from new terminal command:
+## Writing Tests
+
+### Basic Test Structure
+
+Playwright tests follow a simple structure. Each test case is a JavaScript or TypeScript function with Playwright APIs for interacting with the browser.
+
+```javascript
+const { test } = require("@playwright/test");
+test("Example Test", async ({ page }) => {
+  // Your test logic goes here
+});
+```
+
+### Locators
+
+#### Prefer User-Facing Attributes
+
+When selecting elements, prefer using user-facing attributes over XPath or CSS selectors. This ensures that the tests are tied to the actual user experience and are less prone to breakage when the underlying structure changes.
+
+```javascript
+// Bad: Using XPath
+await page.click('//div[@class="my-button"]');
+// Good: Using User-Facing Attribute
+await page.click('[data-testid="my-button"]');
+```
+
+#### Code Generation
+
+To find locators, leverage Playwright's code generation feature. Use the following command to record user actions and generate test code:
+
+```bash
+cd client && npx playwright codegen http://localhost:3000
+```
+
+### Best Practices
+
+#### Test Isolation
+
+Ensure that each test is independent and does not rely on the state or outcome of other tests. This helps in maintaining a clean and predictable test suite.
+
+#### Use `expect` Assertions
+
+Use `expect` assertions to verify the expected behavior of the application. This makes your tests more readable and clearly defines the criteria for success.
+
+```javascript
+expect(await page.isVisible('[data-testid="success-message"]')).toBe(true);
+```
+
+#### Playwright Testing Prerequisites
+
+- To run playwright end-to-end tests for the first time, you may need to run `yarn playwright install --with-deps` to install the browsers
+
+**Ensure you have the following configuration**
+
+1. From file `client/e2e/.env.local.example` create `client/e2e/.env.local` with values reflecting instructions in file `client/e2e/.env.local.example`
```

Comment:
In the end we didn't need to do anything for db setup? FYI, Marcel's working on a PR to add more operations, operators, and users, so if that will affect test setup, you might want to chat with him (not sure if he's already merged it)

Reply:
...added Db logic specific to a test in the test. e.g. `client/e2e/pages/onboarding/home/page.spec.ts`